pulp_smash/resources/platform/api_v2/api.py

```diff
@@ -0,0 +1,183 @@
+        """
+        self.last_response = requests.get(
+            self.cfg.base_url + REPOSITORY_PATH.format(self.data_keys['id']),
+            **self.cfg.get_requests_kwargs()
+        )
+
+    def update_repo(self,
+                    delta,
+                    importer_config=None,
+                    distributor_configs=None):
+        """Update repository with keys from kwargs.
+        After calling this method, <repo>.last_response.raise_for_status()
+        and Task.wait_for_tasks(<repo>.last_response)
+        should be called in order to make sure repo was correctly updated.
+        :param delta: Object containing keys with values that should
+            be updated on the repository.
+        :param importer_config: Object containing keys with values that should
+            be updated on the repository’s importer config.
+        :param distributor_configs: object containing keys that
+            are distributor ids
+        """
+        my_delta = {'delta': delta}
+        if importer_config is not None:
+            my_delta.update({'importer_config': importer_config})
+        if distributor_configs is not None:
+            my_delta.update({'distributor_configs': distributor_configs})
+        self.last_response = requests.put(
+            self.cfg.base_url + REPOSITORY_PATH.format(self.data_keys['id']),
+            json=my_delta,
+            **self.cfg.get_requests_kwargs()
+        )
+
+
+class Task(object):
+    """Handles tasks related operations. So far only waiting for given tasks
+    to immediate finish is implemented.
+    """
+    cfg = get_config()
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def _wait_for_task(cls, task, timeout, frequency):
+        """Wait for single task to finish its execution on server.
+        :param task: Dictionary containtin task_id and path to task
+            on pulp server.
+        :param timeout: Timeout in seconds for each task to complete.
+        :param frequency: Task polling frequency in seconds.
+        """
+        task_timeout = time.time() + timeout
+        while time.time() <= task_timeout:
+            time.sleep(frequency)
+            response = requests.get(
+                Task.cfg.base_url +
+                POLL_TASK_PATH.format(task["task_id"]),
+                **Task.cfg.get_requests_kwargs()
+            )
+            try:
+                response.raise_for_status()
+            except HTTPError:
+                break
+            # task finished with error
+            if response.json()["state"] in TASK_ERROR_STATES:
+                raise Exception("Error occured while polling task: ",
+                                response.text)
+            # task finished properly
+            if response.json()["state"] in TASK_FINISHED_STATES:
+                break
+        # task probably timed out
+        else:
+            raise Exception("Timeout occured while waiting for task: ",
```

Comment:
Never raise `Exception`. It's a Bad Thing™ to do. If a client wants to catch this exception, they will write something like this:

```python
try:
    Task.wait_for_task()
except Exception:
    pass # do something
```

This is problematic because Exception is the parent class of most exceptions. This code might catch completely unrelated exceptions.